cfme/tests/test_utilization_metrics.py

```diff
@@ -151,28 +154,48 @@ def test_raw_metric_vm_cpu(metrics_collection, db, provider):
             vm_name)
 
     for record in query:
-        if record.cpu_usagemhz_rate_average is None:
-            pass
-        else:
-            assert record.cpu_usagemhz_rate_average > 0, 'Zero VM CPU Usage'
-            break
+        if provider.type == 'infra':
+            if record.cpu_usagemhz_rate_average is None:
+                pass
+            else:
+                assert record.cpu_usagemhz_rate_average > 0, 'Zero VM CPU Usage'
+                break
```

Comment:
In addition, to make it even shorter and less repetitive, you could do:
```
from operator import attrgetter

if provider.category == "infra":
    query = query_metric_db(db, provider, 'cpu_usagemhz_rate_average',
             vm_name)
    average_rate = attrgetter('record.cpu_usagemhz_rate_average')
elif provider.category == "cloud":
    query = query_metric_db(db, provider, 'cpu_usage_rate_average',
              vm_name)
    average_rate = attrgetter('cpu_usage_rate_average')
```

Now it is split by provider category all in one conditional so you can do just:
```
for record in query:
    if average_rate(record) is not None:
        assert average_rate(record) > 0, 'Zero VM CPU Usage'
```
and that's it :)

The attrgetter thing is really what the name says - its an attribute getter. You just need to tell it what kind of attribute it should be fetching and then give it an object, in our case a record, and it will fetch that given attribute.